front-end/generate-task-lists.js

```diff
@@ -62,13 +64,39 @@ const renderTaskList = event => {
 
             const markTaskAsDone = event => {
                 task.status = STATUSES.DONE;
```

Comment:
Avoid mutating objects/arrays directly, because it can cause some unexpected issues. Actually there's no need to set a new status before making request that writes the change to the database, because what happens when this request fails? 

So just replace the `status` line in the axios call with the value of `STATUSES.DONE` and let the server handle this work.